source/slang/slang-ir-generics-lowering-context.h

```diff
@@ -31,6 +31,9 @@ namespace Slang
         // Dictionaries for interface type requirement key-value lookups.
         // Used by `findInterfaceRequirementVal`.
         Dictionary<IRInterfaceType*, Dictionary<IRInst*, IRInst*>> mapInterfaceRequirementKeyValue;
+        
+        // Map from interface requirement keys to its corresponding dispatch method.
+        OrderedDictionary<IRInst*, IRFunc*> mapInterfaceRequirementKeyToDispatchMethods;
```

Comment:
Eventually we need to figure out how this will work in the presence of generic interfaces. If you have:

```
interface IHolder<T>
{
    T getValue();
}
```

Then there will be a single key for `IHolder<...>::getValue`, and even if we do explicit specialization, that key won't change. That means that witness tables for `IHolder<int>` and `IHolder<double>` would use the same key.

As a result, it seems like we have two choices here:

1. Ensure that the dispatch method we generate is usable for any instantiation of a generic interface. That would seem to require that the dispatch method would itself be generic (e.g., `IHolder_getValue<T>(...)`), or be equivalent to the dynamic-dispatch version of such a generic.

2. Generate distinct dispatch methods for each instantiation of `IHolder<...>`. In that case, it seems like the `mapInterfaceRequirementKeyToDispatchMethod` approach wouldn't be enough, since it would need to map the pair of `(specializedInterface, requirementKey)` to the dispatch method.

Naively, it seems to me like option (1) is closest to what we'd want long-term, but you'd probably have a better intuition given that you've implemented most of the logic involved. Thoughts?

Reply:
None of the other parts of `lower-generics` pass considers/handles the possibility of generic interfaces. I think one possible answer is to have a pre-pass that turns all generic interfaces into interfaces with generic methods, and then lower those generic methods as usual.